packages/grafana-data/src/utils/dataLinks.ts

```diff
@@ -64,6 +64,7 @@ export function mapInternalLinkToExplore(options: LinkToExploreOptions): LinkMod
       : undefined,
     target: link?.targetBlank ? '_blank' : '_self',
     origin: field,
+    wasQueryInterpolated: JSON.stringify(interpolatedQuery) !== JSON.stringify(link.internal?.query),
```

Comment:
This seems to be the best way to globally ensure the link had a change to it and thus the log line should show up.

Reply:
It may happen that only one variable gets interpolated. In such case, `wasQueryInterpolated` would be true, but in fact, the query wouldn't be fully interpolated.

I think we could try get all variables out of the query and check if they still exists after interpolation. This is what was added specifically to TraceView - https://github.com/grafana/grafana/pull/61702/files#diff-d0224e7cbd216eae15f014f214d953a9774cf564e15e825a10f93bb92c20ac0fR191. It's not covering all cases but maybe a good start.